test/acceptance/tests/basic_test.go

```diff
@@ -55,7 +60,9 @@ func TestBasic(t *testing.T) {
 			partition := ""
 			queryString := ""
 			tfVars["consul_image"] = "public.ecr.aws/hashicorp/consul:1.15.1"
-
+			if c.autoPublishRegistrator {
+				tfVars["ecr_image_uri"] = ""
```

Comment:
per my earlier comment about the UX for `var.ecr_image_uri` and `var.private_ecr_repo_name`, I think that in the auto publish case we want to do a `delete(tfVars, "ecr_image_uri")` and set `tfVars["private_ecr_repo_name"]`

Reply:
We will also need to add a cross validation test for `ecr_image_uri` and `private_ecr_repo_name`, similar to what we do in the ECS repo: https://github.com/hashicorp/terraform-aws-consul-ecs/blob/main/test/acceptance/tests/validation/validation_test.go#L169